include/ginkgo/core/base/utils.hpp

```diff
@@ -403,6 +413,7 @@ class copy_back_deleter<const T> {
 
 
 /**
+ * @internal
```

Comment:
I'm not sure about these two `temporary_clone`s.

Reply:
So, should I mark it with internal ?

Reply:
I think as the user can actually call this, I will remove the internal tag.

Reply:
I think so yes, looking at the code they are rather used to ensure Ginkgo works wherever the data might be located with any executor. I think they are in general rather made for that case so I would keep it. I don't know what others think.